packages/react_rearch/lib/src/widgets/consumer.dart

```diff
@@ -0,0 +1,158 @@
+part of '../widgets.dart';
+
+abstract class RearchComponent extends Component2 {
+  bool _building = false;
+
+  String get debugName;
+
+  bool get debug => false;
+
+  void _debug(String msg) {
+    if (!debug) return;
+    // ignore: avoid_print
+    print('<$debugName> -- $msg');
+  }
+
+  final willUnmountListeners = <SideEffectApiCallback>{};
+
+  final sideEffectData = <Object?>[];
+
+  /// Represents a [Set] of functions that remove a dependency on a [Capsule].
+  final dependencyDisposers = <void Function()>{};
```

Comment:
these should probably be prefixed with a `_` so that they won't be visible to subclasses.